### PR TITLE
Recreate hierarchy for a given image

### PR DESCRIPTION
**Pull request by gmiejski**
Let's say you have a given hierarchy of images:
```
python-3.8
    python-3.8-jdk
    python-3.8-web
        python-3.8-flask
```

If you want to quickly introduce python3.9 for those images, you can:
1) create `python-3.9` image
2) recreate images hierarchy

```
docker-bakery copy-images-hierarchy --config ${path_to_config} --base-image sr-python3.8 -rp 3.8=3.9 -r   
```

The resulting hierarchy will look like this:

```
python-3.9
    python-3.9-jdk
    python-3.9-web
        python-3.9-flask
```

For more options run `docker-bakery copy-images-hierarchy help`

### CODE REVIEW

File: bakery/service/update_generation.go [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+	return replacer
+}
+
+func generateAncestors(previousImageName string, dependenciesMap map[string][]*DockerImage, replacer nameReplacer, recursive bool, skipExistingDirectories bool) error {
+	ancestorsToGenerate, err := getChildrenToUpdate(previousImageName, dependenciesMap, replacer, recursive)
+	fmt.Println("Found ancestors to recreate:")
+	for _, ancestor := range ancestorsToGenerate {
+		fmt.Println(ancestor.childOriginImageName)
+	}
+	if err != nil {
```
Comment (kasinskim) [inline bakery/service/update_generation.go:43]: Shouldn't you check that error presence before processing `ancestorsToGenerate` ?

File: bakery/service/update_generation.go [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+			continue
+		}
+		replacer = replacer.add(keyValuePair[0], keyValuePair[1])
+	}
+	return replacer
+}
+
+func generateAncestors(previousImageName string, dependenciesMap map[string][]*DockerImage, replacer nameReplacer, recursive bool, skipExistingDirectories bool) error {
+	ancestorsToGenerate, err := getChildrenToUpdate(previousImageName, dependenciesMap, replacer, recursive)
+	fmt.Println("Found ancestors to recreate:")
```
Comment (kasinskim) [inline bakery/service/update_generation.go:39]: I'd suggest extracting `printAncestors` function just for a sake of better readability

File: bakery/service/update_generation.go [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+	}
+}
+
+type nameReplacer struct {
+	replacements []changeNameFn
+}
+
+func (n nameReplacer) add(from, to string) nameReplacer {
+	n.replacements = append(n.replacements, simpleNameReplacer(from, to))
+	from2 := strings.ToUpper(strings.Replace(from, "-", "_", -1))
```
Comment (kasinskim) [inline bakery/service/update_generation.go:107]: Could be extracted to separate function with meaningful name (e.g. `buildImageVersionVariable`).
```
	strings.ToUpper(strings.Replace(from, "-", "_", -1))
```

File: bakery/service/update_generation.go (hunk 1 above)
Comment (gmiejski) [reply bakery/service/update_generation.go:43]: The code would still work, as nil is a valid thing to iterate over, but it's more elegant to do as you said - changed 😄

File: bakery/service/update_generation.go (hunk 3 above)
Comment (gmiejski) [reply bakery/service/update_generation.go:107]: done

**Review by kasinskim — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -213,3 +213,32 @@ Applying `docker-bakery` is quite simple. Take a look [here](https://github.com/
 <a id="limitations"></a>
 # Limitations
 At the moment docker multi-stage builds are not fully supported. Only the first line `FROM` is taken into consideration when determining the parent of the image and its place in the hierarchy. 
+
+## Command copy-images-hierarchy
```
Comment (jaroslaw-bochniak) [inline README.md:217]: Description of this command should not be a subsection of #Limitations.
Also the TOC table should be recreated.

Comment (gmiejski) [reply README.md:211]: done

**Review by jaroslaw-bochniak — APPROVED** [review]

**Pull request merged**